contracts/IdentityFactory.sol

```diff
@@ -8,26 +8,59 @@ contract IdentityFactory is Initializable {
 
   event IdentityCreated(address indexed identity);
 
+  /**
+  * @dev Keep track or identities created with the factory contract
+  * This is necessary because there is no way to tell on chain
+  * if the identity interface is implemented in a certain way and
+  * the Centrifuge Protocol requires on chain key history. A key
+  * should be revoked and not removed
+  */
+  mapping(address => bool) internal _identities;
+
   /**
   * Deploys a new identity and transfers the ownership to the sender
   */
   function createIdentity()
   external
   {
-    Identity identity_ = new Identity(msg.sender);
-    emit IdentityCreated(address(identity_));
+
+    Identity identity_ = new Identity(msg.sender, new bytes32[](0), new uint256[](0));
```

Comment:
What is the value of identity_ if the creation of the identity contract fails? (Let's say not sufficient gas?, stackoverflow etc.)

Is it 0x?
In that case we shouldn't throw an IdentityCreated event and not add it to the list.
Maybe gas is a bad example.

Reply:
I am assuming that the transaction will fail. How would you test that?

Reply:
stackoverflow should not happen, not recursive call